src/main/java/pl/javastart/task/VehicleService.java

```diff
@@ -0,0 +1,65 @@
+package pl.javastart.task;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Scanner;
+
+public class VehicleService {
+    static Queue<Vehicle> vehicles = new LinkedList<>();
+    private static String fileName = "vehicles.txt";
+
+    private static Vehicle createVehicle() {
+        Scanner scan = new Scanner(System.in);
+        System.out.println("Podaj informacje o pojeździe: typ, marka, model, rocznik, przebieg i numer VIN");
+        String type = scan.nextLine();
+        String brand = scan.nextLine();
+        String model = scan.nextLine();
+        int year = scan.nextInt();
+        int mileage = scan.nextInt();
+        scan.nextLine();
+        String vin = scan.nextLine();
+        return new Vehicle(type, brand, model, year, mileage, vin);
+    }
+
+    static Queue<Vehicle> addVehicle() {
+        if (!fileName.isEmpty()) {
+            readSavedVehicles();
+        }
+        Vehicle vehicle = createVehicle();
+        vehicles.add(vehicle);
+        return vehicles;
+    }
+
+    static Vehicle readNextVehicle() {
+        Vehicle vehicle = vehicles.poll();
+        System.out.println(vehicle);
+        return vehicle;
+    }
+
+    static void checkQueue() {
+        if (vehicles.isEmpty()) {
+            System.out.println("Koniec");
+        } else {
+            VehicleFiles.saveVehicles(vehicles, fileName);
+        }
+    }
+
+    static Queue<Vehicle> readSavedVehicles() {
+        try {
+            Scanner scan = new Scanner(new File(fileName));
```

Comment:
Pamiętaj o zamykaniu pliku, albo najlepiej wprowadź try-with-resources https://youtu.be/LRwSHA9YDgY